### DISCUSSION TITLE
Use `fancy_regex` instead of `regex` for full-regex support.

### DISCUSSION THREAD

**Discussion by blyxyas** [community/community#51957] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

A lot of times I need to use "advanced" regex features, like backreferences or look-arounds, the problem is that the Rust crate that Github uses for code search ([regex](https://docs.rs/regex)) doesn't and will never support such features. So I suggest having full regex support by using other crates.

In this discussion, Burntsushi (one of `regex`'s mantainers) recommends using `fancy_regex` to support those features.
People (I would think a lot come from this Github search issue) ask for look-around support, but they won't implement it because it isn't the goal of the crate.

Suggestion: Have an "advanced regex search option" which alerts that it may take a little bit longer, or have a "complex-regex detector" and automatically choose between `regex` and `fancy-regex`

---

Relevant to #8653

**Comment by look** [comment, marked as answer]
Thanks for the suggestion. We aren't currently planning to implement this because regular expression features like this have expensive runtimes and that puts our service at risk of denial of service.

**Comment by mike-clark-8192** [comment]
I have already run into the error "your search query was too expensive" with the existing regex engine. It seems there are already safeguards in place for expensive patterns. Would those guards not work the same for fancy_regex?
